src/transformers/models/mamba2/modeling_mamba2.py

```diff
@@ -477,8 +539,10 @@ def torch_forward(self, input_states, cache_params: Optional[Mamba2Cache]=None,
             dBx = dB * hidden_states[..., None]
 
             # State calculation
-            cache_params.ssm_states[self.layer_idx].copy_(
-                cache_params.ssm_states[self.layer_idx] * dA + dBx
+            cache_device = cache_params.ssm_states.device
+            cache_params.update_ssm_state(
+                layer_idx=self.layer_idx,
+                new_ssm_state=cache_params.ssm_states[self.layer_idx] * dA.to(device=cache_device) + dBx.to(device=cache_device)
```

Comment:
Should fix the multi-gpu issues

Reply:
Will update the description if that's the case

Reply:
alright, the idea is to have the move be a no-op if we're on the same GPU, else move it to the correct one? 
I'd be pro-moving explicitly each tensor to its device of choice before, not inline

```suggestion
            cache_device = cache_params.ssm_states.device
            dA = dA.to(device=cache_device)
            dBx = dBx.to(device=cache_device)
            cache_params.update_ssm_state(
                layer_idx=self.layer_idx,
                new_ssm_state=cache_params.ssm_states[self.layer_idx] * dA + dBx
```
and possibly do that even earlier, as soon as the tensors are introduced!

Reply:
changed it a bit and moved to the device as it is created, lmk what you think about it - but multi gpu doesn't seem to like mamba2 either way :D